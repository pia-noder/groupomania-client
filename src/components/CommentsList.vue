<template>
  <div>

    <div  v-for="comment in comments" :key="comment.id" >

      <Comment  
        :comment="comment" 
        :post="post" 
        @delete-comment="pullCommentFromComments"
      />

    </div>

    <button
      v-if="count > 1 && !allCommentsDisplayed"
      @click="getAllComments(post)"
    >

      <span v-if="count > 1">Afficher {{ count - 1 }} autres commentaires</span>
      <span v-else>Afficher {{ count - 1 }} autre commentaire</span>

    </button>

    <CreateComments 
      :user="user" 
      :post="post" 
      @comment-created="pushNewCommentInComments"
      
    />

  </div>
</template>

<script>
import ServiceComments from '@/service/ServiceComments'

import CreateComments from '@/components/CreateComments'
import Comment from '@/components/Comment'


export default {
name: 'CommentsList',
  components: {
    CreateComments,
    Comment,
  },
  props:['post', 'user'],

  created() {
        this.getOneOfAllComments(this.post);
      },

  data () {
        return {
            postIncommentsList:this.post,
            allCommentsDisplayed: false,
            comments:[],
            count:0,
        }
    },
  

  methods:{

    getOneOfAllComments(post) {
      const getFistComment = ServiceComments.getOneOfAllComments(post);

      getFistComment.then((response) => {
        this.comments = response.data[0];
        this.count = response.data[1][0].count;
      })    
    },

    async getAllComments(post){
      ServiceComments.getAllComments(post).then(response => {
        this.comments = response.data;
      });
      this.allCommentsDisplayed = true;
    },

    pushNewCommentInComments(newComment){
      this.comments.push(newComment)
    },
    pullCommentFromComments(deletedComment){
       this.comments = this.comments.filter(comment => comment.comments_id !== deletedComment)
    }
  },
}

</script>

<style>

</style>