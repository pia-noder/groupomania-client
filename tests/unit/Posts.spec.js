import Vue from 'vue'
import Vuex from 'vuex'

import { mount } from '@vue/test-utils'
import CommentsList from '@/components/CommentsList.vue'
import storeData from '@/store/index.js'

Vue.use(Vuex);
//CrÃ©ation du store
const store = new Vuex.Store(storeData);

test('store is loaded', ()=>{
  const wrapper = mount(CommentsList,{
    store
  });
  store.state.user.push({name:'Joe'})
  expect(wrapper.vm.user).toEqual([{name: 'Joe'}])
})
