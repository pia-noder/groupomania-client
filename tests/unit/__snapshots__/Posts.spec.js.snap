// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu is render renders a div 1`] = `
<div class="nav">
  <div class="bloc-top-part"><img id="imgLogoBlack" src="@/assets/icon-left-font-monochrome-black.svg" alt="logo en noir et blanc de Groupomania" class="imgLogoBlack">
    <div class="search">
      <font-awesome-icon icon="search" class="icon-search"></font-awesome-icon> <input aria-label="Faire une recherche" type="text" class="searchBarre">
    </div>
  </div>
  <ul>
    <li><a href="" aria-label="Aller vers page d'accueil" title="Accueil">
        <router-link to="[object Object]">
          <font-awesome-icon icon="home" class="icon fa-lg"></font-awesome-icon>
        </router-link>
      </a></li>
    <li><a href="#" aria-label="Lien vers page du profile" title="Profile utilisateur">
        <router-link to="[object Object]">
          <font-awesome-icon icon="user-circle" class="icon fa-lg"></font-awesome-icon>
        </router-link>
      </a></li>
    <li><a href="" aria-label="Notifications" title="Notification">
        <router-link to="[object Object]" class="link">
          <font-awesome-icon icon="bell" class="icon fa-lg"></font-awesome-icon>
        </router-link>
      </a></li>
    <li><a href="" aria-label="Newpaper" title="Newpaper">
        <router-link to="[object Object]" class="link">
          <font-awesome-icon icon="newspaper" class="icon fa-lg"></font-awesome-icon>
        </router-link>
      </a></li>
    <li><a href="" aria-label="Déconnexion" title="déconnexion">
        <router-link to="[object Object]">
          <font-awesome-icon icon="sign-out-alt" class="icon fa-lg"></font-awesome-icon>
        </router-link>
      </a></li>
  </ul>
</div>
`;
